DuckDuckGo/Autofill/AutofillCredentialsImportManager.swift

```diff
@@ -24,18 +24,31 @@ public protocol AutofillCredentialsImportPresentationDelegate: AnyObject {
     func autofillDidRequestCredentialsImportFlow(onFinished: @escaping () -> Void, onCancelled: @escaping () -> Void)
 }
 
+public protocol AutofillLoginImportStateProvider {
+    var isEligibleDDGUser: Bool { get }
+    var hasImportedLogins: Bool { get }
+    var isAutofillEnabled: Bool { get }
+    var isCredentialsImportPromptPermanantlyDismissed: Bool { get }
+    func hasNeverPromptWebsitesFor(_ domain: String) -> Bool
+}
+
 final public class AutofillCredentialsImportManager {
-    private let userDefaults: UserDefaults
+    private var stateStore: AutofillLoginImportStateStoring
+    private var loginImportStateProvider: AutofillLoginImportStateProvider
+    private let isBurnerWindow: Bool
 
     weak var presentationDelegate: AutofillCredentialsImportPresentationDelegate?
 
-    init(userDefaults: UserDefaults = .standard) {
-        self.userDefaults = userDefaults
+    init(stateStore: AutofillLoginImportStateStoring = AutofillLoginImportState(),
```

Comment:
I think `loginImportStateProvider` can be used instead of `stateStore` since it also contains `isCredentialsImportPromptPermanantlyDismissed`?